dokka-subprojects/analysis-kotlin-descriptors/build.gradle.kts

```
@@ -47,5 +47,34 @@ tasks.shadowJar {
  *   and so such classes are not replaced afterward by `shadowJar` task - it visits every class once
  *
  */
-dependencies.shadow(libs.fastutil)
-tasks.shadowJar { exclude("it/unimi/dsi/fastutil/**") }
+
+// configuration for dependencies which we need to replace with original ones because `kotlin-compiler` minimizes them
+val shadowOverride: Configuration by configurations.creating {
+    attributes {
+        attribute(Usage.USAGE_ATTRIBUTE, project.objects.named(Usage::class.java, "java-runtime"))
```

```suggestion
        attribute(Usage.USAGE_ATTRIBUTE, project.objects.named("java-runtime"))
```

use kotlin-dsl overload for `ObjectFactor.named()`

---

Thanks!

---

Updated manually to fix some other places